Dockerfile 修复下载链接错误

Dimitri2020007 opened the pull request.

eddie-chiang commented:
这PR貌似会出现一下问题：
```
=== Fetching app code
=== Building web (Dockerfile)
=!= Dockerfile parse error line 25: unknown instruction: CURL
```

Dimitri2020007 commented:
> 这PR貌似会出现以下问题：
> 
> ```
> === Fetching app code
> === Building web (Dockerfile)
> =!= Dockerfile parse error line 25: unknown instruction: CURL
> ```

你能顺着我刚刚改的commit再次测试下吗？

我的xrayku被ban了试不出来

eddie-chiang commented:
```apk del .build-deps``` line 28 这行可能不需要。
```
=== Fetching app code
=== Building web (Dockerfile)
Sending build context to Docker daemon  284.7kBStep 1/10 : FROM caddy:builder-alpine AS builder
builder-alpine: Pulling from library/caddy
29291e31a76a: Pulling fs layer
e4bc8fc554c3: Pulling fs layer
803daa35ea47: Pulling fs layer
de40ce539e25: Pulling fs layer
b1d2e2be38c8: Pulling fs layer
6c16d3fa0b8b: Pulling fs layer
679f5a4b4a2e: Pulling fs layer
9177def7cb28: Pulling fs layer
b1d2e2be38c8: Waiting
6c16d3fa0b8b: Waiting
679f5a4b4a2e: Waiting
9177def7cb28: Waiting
de40ce539e25: Waiting
803daa35ea47: Verifying Checksum
803daa35ea47: Download complete
29291e31a76a: Verifying Checksum
29291e31a76a: Download complete
e4bc8fc554c3: Verifying Checksum
e4bc8fc554c3: Download complete
b1d2e2be38c8: Verifying Checksum
b1d2e2be38c8: Download complete
6c16d3fa0b8b: Verifying Checksum
6c16d3fa0b8b: Download complete
29291e31a76a: Pull complete
679f5a4b4a2e: Verifying Checksum
679f5a4b4a2e: Download complete
9177def7cb28: Verifying Checksum
9177def7cb28: Download complete
e4bc8fc554c3: Pull complete
803daa35ea47: Pull complete
de40ce539e25: Verifying Checksum
de40ce539e25: Download complete
de40ce539e25: Pull complete
b1d2e2be38c8: Pull complete
6c16d3fa0b8b: Pull complete
679f5a4b4a2e: Pull complete
9177def7cb28: Pull complete
Digest: sha256:8f954fecc9768db82118ff53b67fe90116a757dae5daaac003105f000712455c
Status: Downloaded newer image for caddy:builder-alpine
 ---> f64009d1a35a
Step 2/10 : RUN xcaddy build         --with github.com/mholt/caddy-l4         --with github.com/mholt/caddy-dynamicdns         --with github.com/caddy-dns/openstack-designate         --with github.com/caddy-dns/azure         --with github.com/caddy-dns/vultr         --with github.com/caddy-dns/hetzner         --with github.com/caddy-dns/digitalocean         --with github.com/caddy-dns/alidns         --with github.com/caddy-dns/gandi         --with github.com/caddy-dns/duckdns         --with github.com/caddy-dns/dnspod         --with github.com/caddy-dns/lego-deprecated         --with github.com/caddy-dns/route53         --with github.com/caddy-dns/cloudflare
 ---> Running in 665f943cd013
2021/08/22 05:36:50 [INFO] Temporary folder: /tmp/buildenv_2021-08-22-0536.187304441
2021/08/22 05:36:50 [INFO] Writing main module: /tmp/buildenv_2021-08-22-0536.187304441/main.go
2021/08/22 05:36:50 [INFO] Initializing Go module
2021/08/22 05:36:50 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init caddy 
go: creating new go.mod: module caddy
go: to add module requirements and sums:
	go mod tidy
2021/08/22 05:36:50 [INFO] Pinning versions
2021/08/22 05:36:50 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddyserver/caddy/v2@v2.4.3 
go: downloading github.com/caddyserver/caddy/v2 v2.4.3
go: downloading github.com/caddyserver/certmagic v0.14.0
go: downloading github.com/google/uuid v1.2.0
go: downloading github.com/prometheus/client_golang v1.10.1-0.20210603120351-253906201bda
go: downloading go.uber.org/zap v1.17.0
go: downloading golang.org/x/term v0.0.0-20210503060354-a79de5458b56
go: downloading github.com/klauspost/cpuid v1.2.0
go: downloading github.com/libdns/libdns v0.2.1
go: downloading github.com/mholt/acmez v0.1.3
go: downloading github.com/miekg/dns v1.1.42
go: downloading github.com/klauspost/cpuid/v2 v2.0.6
go: downloading golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a
go: downloading golang.org/x/net v0.0.0-20210525063256-abc453219eb5
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/common v0.26.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading golang.org/x/sys v0.0.0-20210603081109-ebe580a85c40
go: downloading go.uber.org/atomic v1.7.0
go: downloading go.uber.org/multierr v1.6.0
go: downloading golang.org/x/text v0.3.6
go: downloading google.golang.org/protobuf v1.26.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go get: added github.com/caddyserver/caddy/v2 v2.4.3
2021/08/22 05:37:04 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/mholt/caddy-l4 
go: downloading github.com/mholt/caddy-l4 v0.0.0-20210604203514-3cfcafe70883
go: downloading github.com/mastercactapus/proxyprotocol v0.0.3
go: downloading golang.org/x/time v0.0.0-20201208040808-7e3f01d25324
go: downloading github.com/google/cel-go v0.7.3
go: downloading github.com/lucas-clemente/quic-go v0.21.0
go: downloading google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c
go: downloading github.com/dustin/go-humanize v1.0.1-0.20200219035652-afde56e7acac
go: downloading github.com/klauspost/cpuid v1.2.5
go: downloading github.com/smallstep/certificates v0.15.15
go: downloading github.com/stoewer/go-strcase v1.2.0
go: downloading github.com/marten-seemann/qpack v0.2.1
go: downloading github.com/smallstep/cli v0.15.16
go: downloading github.com/smallstep/truststore v0.9.6
go: downloading github.com/pkg/errors v0.9.1
go: downloading go.step.sm/crypto v0.8.0
go: downloading github.com/antlr/antlr4 v0.0.0-20200503195918-621b933c7a7f
go: downloading github.com/cheekybits/genny v1.0.0
go: downloading github.com/marten-seemann/qtls-go1-15 v0.1.4
go: downloading github.com/marten-seemann/qtls-go1-16 v0.1.3
go: downloading github.com/marten-seemann/qtls-go1-17 v0.1.0-alpha.1
go: downloading github.com/smallstep/nosql v0.3.6
go: downloading howett.net/plist v0.0.0-20200419221736-3b63eb3a43b5
go: downloading gopkg.in/square/go-jose.v2 v2.5.1
go: downloading github.com/Masterminds/sprig/v3 v3.2.2
go: downloading go.step.sm/cli-utils v0.2.0
go: downloading github.com/urfave/cli v1.22.4
go: downloading github.com/Masterminds/goutils v1.1.1
go: downloading github.com/Masterminds/semver v1.4.2
go: downloading github.com/huandu/xstrings v1.3.1
go: downloading github.com/Masterminds/semver/v3 v3.1.1
go: downloading github.com/imdario/mergo v0.3.11
go: downloading github.com/mitchellh/copystructure v1.0.0
go: downloading github.com/shopspring/decimal v1.2.0
go: downloading github.com/spf13/cast v1.3.1
go: downloading github.com/dgraph-io/badger v1.6.2
go: downloading go.etcd.io/bbolt v1.3.5
go: downloading github.com/dgraph-io/badger/v2 v2.0.1-rc1.0.20201003150343-5d1bab4fc658
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e
go: downloading github.com/manifoldco/promptui v0.8.0
go: downloading github.com/cpuguy83/go-md2man v1.0.10
go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.0
go: downloading github.com/mitchellh/reflectwalk v1.0.0
go: downloading github.com/dgraph-io/ristretto v0.0.4-0.20200906165740-41ebdbffecfd
go: downloading github.com/dgryski/go-farm v0.0.0-20200201041132-a6ae2369ad13
go: downloading github.com/samfoo/ansi v0.0.0-20160124022901-b6bd2ded7189
go: downloading github.com/juju/ansiterm v0.0.0-20180109212912-720a0952cc2a
go: downloading github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96
go: downloading github.com/russross/blackfriday v1.5.2
go: downloading github.com/golang/snappy v0.0.3
go: downloading github.com/DataDog/zstd v1.4.5
go: downloading github.com/russross/blackfriday/v2 v2.0.1
go: downloading github.com/shurcooL/sanitized_anchor_name v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.6
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/lunixbochs/vtclean v1.0.0
go get: added github.com/mholt/caddy-l4 v0.0.0-20210604203514-3cfcafe70883
2021/08/22 05:37:09 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/mholt/caddy-dynamicdns 
go: downloading github.com/mholt/caddy-dynamicdns v0.0.0-20210719141002-2110627e7fea
go: downloading github.com/NebulousLabs/go-upnp v0.0.0-20181203152547-b32978b8ccbf
go: downloading gitlab.com/NebulousLabs/fastrand v0.0.0-20181126182046-603482d69e40
go: downloading gitlab.com/NebulousLabs/go-upnp v0.0.0-20181011194642-3a71999ed0d3
go get: added github.com/mholt/caddy-dynamicdns v0.0.0-20210719141002-2110627e7fea
2021/08/22 05:37:11 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/openstack-designate 
go: downloading github.com/caddy-dns/openstack-designate v0.1.0
go: downloading github.com/libdns/openstack-designate v0.1.0
go: downloading github.com/gophercloud/gophercloud v0.15.0
go get: added github.com/caddy-dns/openstack-designate v0.1.0
2021/08/22 05:37:12 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/azure 
go: downloading github.com/caddy-dns/azure v0.2.0
go: downloading github.com/libdns/azure v0.2.0
go: downloading github.com/Azure/azure-sdk-for-go v52.4.0+incompatible
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/Azure/go-autorest/autorest v0.11.17
go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.7
go: downloading github.com/Azure/go-autorest/autorest/to v0.4.0
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.11
go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.2
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading github.com/dimchansky/utfbom v1.1.1
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/form3tech-oss/jwt-go v3.2.2+incompatible
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/Azure/azure-sdk-for-go v0.2.0-beta
go: downloading github.com/form3tech-oss/jwt-go v1.0.2
go get: added github.com/caddy-dns/azure v0.2.0
2021/08/22 05:37:19 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/vultr 
go: downloading github.com/caddy-dns/vultr v0.0.0-20210105121117-3162aa6b9c27
go: downloading github.com/libdns/vultr v0.0.0-20201128180404-1d5ee21ea62f
go: downloading github.com/vultr/govultr/v2 v2.0.0
go: downloading golang.org/x/oauth2 v0.0.0-20201109201403-9fd604954f58
go: downloading github.com/google/go-querystring v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.6.8
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading google.golang.org/appengine v1.6.7
go get: added github.com/caddy-dns/vultr v0.0.0-20210105121117-3162aa6b9c27
2021/08/22 05:37:20 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/hetzner 
go: downloading github.com/caddy-dns/hetzner v0.0.1
go: downloading github.com/libdns/hetzner v0.0.1
go get: added github.com/caddy-dns/hetzner v0.0.1
2021/08/22 05:37:20 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/digitalocean 
go: downloading github.com/caddy-dns/digitalocean v0.0.0-20210809220558-ac6e4fd9e135
go: downloading github.com/libdns/digitalocean v0.0.0-20210310230526-186c4ebd2215
go: downloading github.com/digitalocean/godo v1.41.0
go get: added github.com/caddy-dns/digitalocean v0.0.0-20210809220558-ac6e4fd9e135
2021/08/22 05:37:21 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/alidns 
go: downloading github.com/caddy-dns/alidns v1.0.23
go: downloading github.com/libdns/alidns v1.0.2-x2
go get: added github.com/caddy-dns/alidns v1.0.23
2021/08/22 05:37:22 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/gandi 
go: downloading github.com/caddy-dns/gandi v1.0.2
go: downloading github.com/libdns/gandi v1.0.2
go get: added github.com/caddy-dns/gandi v1.0.2
2021/08/22 05:37:22 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/duckdns 
go: downloading github.com/caddy-dns/duckdns v0.3.1
go: downloading github.com/libdns/duckdns v0.1.1
go get: added github.com/caddy-dns/duckdns v0.3.1
2021/08/22 05:37:23 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/dnspod 
go: downloading github.com/caddy-dns/dnspod v0.0.3
go: downloading github.com/libdns/dnspod v0.0.3
go: downloading github.com/nrdcg/dnspod-go v0.4.0
go get: added github.com/caddy-dns/dnspod v0.0.3
2021/08/22 05:37:23 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/lego-deprecated 
go: downloading github.com/caddy-dns/lego-deprecated v0.0.0-20210524165322-3ab114dafc6b
go: downloading github.com/go-acme/lego/v4 v4.3.1
go: downloading github.com/cpu/goacmedns v0.1.1
go: downloading github.com/aliyun/alibaba-cloud-sdk-go v1.61.976
go: downloading github.com/nrdcg/auroradns v1.0.1
go: downloading github.com/labbsr0x/bindman-dns-webhook v1.0.2
go: downloading github.com/cloudflare/cloudflare-go v0.14.0
go: downloading github.com/nrdcg/desec v0.5.0
go: downloading github.com/gophercloud/gophercloud v0.16.0
go: downloading github.com/gophercloud/utils v0.0.0-20210216074907-f6de111f2eae
go: downloading github.com/dnsimple/dnsimple-go v0.63.0
go: downloading github.com/akamai/AkamaiOPEN-edgegrid-golang v1.1.0
go: downloading github.com/exoscale/egoscale v0.46.0
go: downloading cloud.google.com/go v0.70.0
go: downloading google.golang.org/api v0.33.0
go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
go: downloading github.com/iij/doapi v0.0.0-20190504054126-0bbf12d6d7df
go: downloading github.com/cenkalti/backoff/v4 v4.1.0
go: downloading github.com/nrdcg/goinwx v0.8.1
go: downloading github.com/pquerna/otp v1.3.0
go: downloading github.com/aws/aws-sdk-go v1.37.27
go: downloading github.com/linode/linodego v0.25.3
go: downloading github.com/liquidweb/liquidweb-go v1.6.3
go: downloading github.com/namedotcom/go v0.0.0-20180403034216-08470befbe04
go: downloading github.com/nrdcg/namesilo v0.2.1
go: downloading gopkg.in/ns1/ns1-go.v2 v2.4.4
go: downloading github.com/oracle/oci-go-sdk v24.3.0+incompatible
go: downloading github.com/ovh/go-ovh v1.1.0
go: downloading github.com/sacloud/libsacloud v1.36.2
go: downloading github.com/transip/gotransip/v6 v6.6.0
go: downloading github.com/OpenDNS/vegadns2client v0.0.0-20180418235048-a3fa4a771d87
go: downloading github.com/vultr/govultr/v2 v2.4.0
go: downloading github.com/Azure/go-autorest/autorest/validation v0.1.0
go: downloading github.com/labbsr0x/goh v1.0.1
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/patrickmn/go-cache v2.1.0+incompatible
go: downloading github.com/sirupsen/logrus v1.6.0
go: downloading gopkg.in/ini.v1 v1.62.0
go: downloading github.com/gofrs/uuid v3.2.0+incompatible
go: downloading google.golang.org/grpc v1.38.0
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/kolo/xmlrpc v0.0.0-20200310150728-e0350524596b
go: downloading github.com/mitchellh/mapstructure v1.3.3
go: downloading github.com/liquidweb/go-lwApi v0.0.5
go: downloading github.com/go-resty/resty/v2 v2.1.1-0.20191201195748-d7b97669fe48
go: downloading go.uber.org/ratelimit v0.0.0-20180316092928-c15da0234277
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading github.com/json-iterator/go v1.1.11
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/go-errors/errors v1.0.1
go: downloading github.com/konsorten/go-windows-terminal-sequences v1.0.3
go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading github.com/boombuler/barcode v1.0.1-0.20190219062509-6c824513bacc
go: downloading github.com/liquidweb/liquidweb-cli v0.6.9
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/deepmap/oapi-codegen v1.3.11
go: downloading github.com/jarcoal/httpmock v1.0.6
go: downloading github.com/stretchr/testify v1.7.0
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/stretchr/objx v0.3.0
go: downloading github.com/k0kubun/go-ansi v0.0.0-20180517002512-3bf9e2903213
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/gofrs/uuid v1.2.0
go: downloading github.com/oracle/oci-go-sdk v1.8.0
go: downloading github.com/patrickmn/go-cache v1.0.0
go get: added github.com/caddy-dns/lego-deprecated v0.0.0-20210524165322-3ab114dafc6b
2021/08/22 05:37:37 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/route53 
go: downloading github.com/caddy-dns/route53 v1.1.2
go: downloading github.com/libdns/route53 v1.1.1
go get: added github.com/caddy-dns/route53 v1.1.2
2021/08/22 05:37:38 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/cloudflare 
go: downloading github.com/caddy-dns/cloudflare v0.0.0-20210607183747-91cf700356a1
go: downloading github.com/libdns/cloudflare v0.1.0
go get: added github.com/caddy-dns/cloudflare v0.0.0-20210607183747-91cf700356a1
2021/08/22 05:37:38 [INFO] Build environment ready
2021/08/22 05:37:38 [INFO] Building Caddy
2021/08/22 05:37:38 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy 
go: downloading github.com/google/go-cmp v0.5.5
go: downloading github.com/go-chi/chi v4.1.2+incompatible
go: downloading gopkg.in/natefinch/lumberjack.v2 v2.0.0
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/klauspost/compress v1.13.0
go: downloading github.com/smallstep/assert v0.0.0-20200723003110-82e2b9b3b262
go: downloading github.com/alecthomas/chroma v0.9.1
go: downloading github.com/naoina/toml v0.1.1
go: downloading github.com/yuin/goldmark v1.3.7
go: downloading github.com/yuin/goldmark-highlighting v0.0.0-20210516132338-9216f9c5aa01
go: downloading github.com/BurntSushi/toml v0.3.1
go: downloading gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15
go: downloading github.com/Azure/go-autorest/autorest/mocks v0.4.1
go: downloading gopkg.in/h2non/gock.v1 v1.0.15
go: downloading github.com/hashicorp/go-hclog v0.9.2
go: downloading github.com/golang/mock v1.5.0
go: downloading github.com/onsi/ginkgo v1.14.0
go: downloading github.com/onsi/gomega v1.10.1
go: downloading github.com/urfave/cli v1.22.5
go: downloading github.com/naoina/go-stringutil v0.1.0
go: downloading github.com/kylelemons/godebug v1.1.0
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/alecthomas/assert v0.0.0-20170929043011-405dbfeb8e38
go: downloading github.com/kr/pretty v0.2.1
go: downloading github.com/smartystreets/goconvey v1.6.4
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/uber-go/atomic v1.3.2
go: downloading github.com/jmespath/go-jmespath/internal/testify v1.5.1
go: downloading github.com/h2non/parth v0.0.0-20190131123155-b4df798d6542
go: downloading github.com/dlclark/regexp2 v1.4.0
go: downloading github.com/alecthomas/colour v0.0.0-20160524082231-60882d9e2721
go: downloading github.com/alecthomas/repr v0.0.0-20180818092828-117648cd9897
go: downloading github.com/sergi/go-diff v1.0.0
go: downloading github.com/kr/text v0.1.0
go: downloading github.com/jtolds/gls v4.20.0+incompatible
go: downloading github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d
go: downloading github.com/nxadm/tail v1.4.4
go: downloading github.com/chzyer/test v0.0.0-20180213035817-a1ea475d72b1
go: downloading github.com/danwakefield/fnmatch v0.0.0-20160403171240-cbb64ac3d964
go: downloading github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/chzyer/logex v1.1.10
go: downloading github.com/OneOfOne/xxhash v1.2.2
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/fsnotify/fsnotify v1.4.9
2021/08/22 05:37:41 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /usr/bin/caddy -ldflags -w -s -trimpath 
2021/08/22 05:41:04 [INFO] Build complete: ./caddy
2021/08/22 05:41:04 [INFO] Skipping cleanup as requested; leaving folder intact: /tmp/buildenv_2021-08-22-0536.187304441
Removing intermediate container 665f943cd013
 ---> 34fe088c60e3
Step 3/10 : FROM caddy:builder-alpine
 ---> f64009d1a35a
Step 4/10 : COPY --from=builder /usr/bin/caddy /usr/bin/caddy
 ---> 488741f3929c
Step 5/10 : RUN apk update &&     apk add --no-cache --virtual ca-certificates caddy tor wget &&     mkdir /v2ray &&     wget -qO- https://github.com/v2fly/v2ray-core/releases/latest/download/v2ray-linux-64.zip | busybox unzip - &&     chmod +x /v2ray &&     rm -rf /var/cache/apk/* &&     apk del .build-deps
 ---> Running in d76662be9822
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
v3.14.1-64-g22f341e695 [https://dl-cdn.alpinelinux.org/alpine/v3.14/main]
v3.14.1-63-gb2aec05acf [https://dl-cdn.alpinelinux.org/alpine/v3.14/community]
OK: 14934 distinct packages available
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
(1/11) Installing caddy (2.4.3-r0)
Executing caddy-2.4.3-r0.pre-install
(2/11) Installing libcap (2.50-r0)
(3/11) Installing libevent (2.1.12-r2)
(4/11) Installing xz-libs (5.2.5-r0)
(5/11) Installing libseccomp (2.5.1-r2)
(6/11) Installing zstd-libs (1.4.9-r1)
(7/11) Installing tor (0.4.5.10-r0)
Executing tor-0.4.5.10-r0.pre-install
(8/11) Installing libunistring (0.9.10-r1)
(9/11) Installing libidn2 (2.3.1-r0)
(10/11) Installing wget (1.21.1-r1)
(11/11) Upgrading ca-certificates (20191127-r5 -> 20210822.054221)
Executing busybox-1.33.1-r3.trigger
OK: 75 MiB in 31 packages
Archive:  -
   creating: systemd/
   creating: systemd/system/
  inflating: systemd/system/v2ray@.service
  inflating: systemd/system/v2ray.service
  inflating: config.json
  inflating: v2ctl
  inflating: geosite.dat
  inflating: vpoint_socks_vmess.json
  inflating: vpoint_vmess_freedom.json
  inflating: geoip.dat
  inflating: v2ray
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/main: No such file or directory
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/community: No such file or directory
ERROR: No such package: .build-deps
The command '/bin/sh -c apk update &&     apk add --no-cache --virtual ca-certificates caddy tor wget &&     mkdir /v2ray &&     wget -qO- https://github.com/v2fly/v2ray-core/releases/latest/download/v2ray-linux-64.zip | busybox unzip - &&     chmod +x /v2ray &&     rm -rf /var/cache/apk/* &&     apk del .build-deps' returned a non-zero code: 1
```

Dimitri2020007 commented:
> `apk del .build-deps` line 28 这行可能不需要。
> 
> ```
> === Fetching app code
> === Building web (Dockerfile)
> Sending build context to Docker daemon  284.7kBStep 1/10 : FROM caddy:builder-alpine AS builder
> builder-alpine: Pulling from library/caddy
> 29291e31a76a: Pulling fs layer
> e4bc8fc554c3: Pulling fs layer
> 803daa35ea47: Pulling fs layer
> de40ce539e25: Pulling fs layer
> b1d2e2be38c8: Pulling fs layer
> 6c16d3fa0b8b: Pulling fs layer
> 679f5a4b4a2e: Pulling fs layer
> 9177def7cb28: Pulling fs layer
> b1d2e2be38c8: Waiting
> 6c16d3fa0b8b: Waiting
> 679f5a4b4a2e: Waiting
> 9177def7cb28: Waiting
> de40ce539e25: Waiting
> 803daa35ea47: Verifying Checksum
> 803daa35ea47: Download complete
> 29291e31a76a: Verifying Checksum
> 29291e31a76a: Download complete
> e4bc8fc554c3: Verifying Checksum
> e4bc8fc554c3: Download complete
> b1d2e2be38c8: Verifying Checksum
> b1d2e2be38c8: Download complete
> 6c16d3fa0b8b: Verifying Checksum
> 6c16d3fa0b8b: Download complete
> 29291e31a76a: Pull complete
> 679f5a4b4a2e: Verifying Checksum
> 679f5a4b4a2e: Download complete
> 9177def7cb28: Verifying Checksum
> 9177def7cb28: Download complete
> e4bc8fc554c3: Pull complete
> 803daa35ea47: Pull complete
> de40ce539e25: Verifying Checksum
> de40ce539e25: Download complete
> de40ce539e25: Pull complete
> b1d2e2be38c8: Pull complete
> 6c16d3fa0b8b: Pull complete
> 679f5a4b4a2e: Pull complete
> 9177def7cb28: Pull complete
> Digest: sha256:8f954fecc9768db82118ff53b67fe90116a757dae5daaac003105f000712455c
> Status: Downloaded newer image for caddy:builder-alpine
>  ---> f64009d1a35a
> Step 2/10 : RUN xcaddy build         --with github.com/mholt/caddy-l4         --with github.com/mholt/caddy-dynamicdns         --with github.com/caddy-dns/openstack-designate         --with github.com/caddy-dns/azure         --with github.com/caddy-dns/vultr         --with github.com/caddy-dns/hetzner         --with github.com/caddy-dns/digitalocean         --with github.com/caddy-dns/alidns         --with github.com/caddy-dns/gandi         --with github.com/caddy-dns/duckdns         --with github.com/caddy-dns/dnspod         --with github.com/caddy-dns/lego-deprecated         --with github.com/caddy-dns/route53         --with github.com/caddy-dns/cloudflare
>  ---> Running in 665f943cd013
> 2021/08/22 05:36:50 [INFO] Temporary folder: /tmp/buildenv_2021-08-22-0536.187304441
> 2021/08/22 05:36:50 [INFO] Writing main module: /tmp/buildenv_2021-08-22-0536.187304441/main.go
> 2021/08/22 05:36:50 [INFO] Initializing Go module
> 2021/08/22 05:36:50 [INFO] exec (timeout=10s): /usr/local/go/bin/go mod init caddy 
> go: creating new go.mod: module caddy
> go: to add module requirements and sums:
> 	go mod tidy
> 2021/08/22 05:36:50 [INFO] Pinning versions
> 2021/08/22 05:36:50 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddyserver/caddy/v2@v2.4.3 
> go: downloading github.com/caddyserver/caddy/v2 v2.4.3
> go: downloading github.com/caddyserver/certmagic v0.14.0
> go: downloading github.com/google/uuid v1.2.0
> go: downloading github.com/prometheus/client_golang v1.10.1-0.20210603120351-253906201bda
> go: downloading go.uber.org/zap v1.17.0
> go: downloading golang.org/x/term v0.0.0-20210503060354-a79de5458b56
> go: downloading github.com/klauspost/cpuid v1.2.0
> go: downloading github.com/libdns/libdns v0.2.1
> go: downloading github.com/mholt/acmez v0.1.3
> go: downloading github.com/miekg/dns v1.1.42
> go: downloading github.com/klauspost/cpuid/v2 v2.0.6
> go: downloading golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a
> go: downloading golang.org/x/net v0.0.0-20210525063256-abc453219eb5
> go: downloading github.com/beorn7/perks v1.0.1
> go: downloading github.com/cespare/xxhash v1.1.0
> go: downloading github.com/cespare/xxhash/v2 v2.1.1
> go: downloading github.com/golang/protobuf v1.5.2
> go: downloading github.com/prometheus/client_model v0.2.0
> go: downloading github.com/prometheus/common v0.26.0
> go: downloading github.com/prometheus/procfs v0.6.0
> go: downloading golang.org/x/sys v0.0.0-20210603081109-ebe580a85c40
> go: downloading go.uber.org/atomic v1.7.0
> go: downloading go.uber.org/multierr v1.6.0
> go: downloading golang.org/x/text v0.3.6
> go: downloading google.golang.org/protobuf v1.26.0
> go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
> go get: added github.com/caddyserver/caddy/v2 v2.4.3
> 2021/08/22 05:37:04 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/mholt/caddy-l4 
> go: downloading github.com/mholt/caddy-l4 v0.0.0-20210604203514-3cfcafe70883
> go: downloading github.com/mastercactapus/proxyprotocol v0.0.3
> go: downloading golang.org/x/time v0.0.0-20201208040808-7e3f01d25324
> go: downloading github.com/google/cel-go v0.7.3
> go: downloading github.com/lucas-clemente/quic-go v0.21.0
> go: downloading google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c
> go: downloading github.com/dustin/go-humanize v1.0.1-0.20200219035652-afde56e7acac
> go: downloading github.com/klauspost/cpuid v1.2.5
> go: downloading github.com/smallstep/certificates v0.15.15
> go: downloading github.com/stoewer/go-strcase v1.2.0
> go: downloading github.com/marten-seemann/qpack v0.2.1
> go: downloading github.com/smallstep/cli v0.15.16
> go: downloading github.com/smallstep/truststore v0.9.6
> go: downloading github.com/pkg/errors v0.9.1
> go: downloading go.step.sm/crypto v0.8.0
> go: downloading github.com/antlr/antlr4 v0.0.0-20200503195918-621b933c7a7f
> go: downloading github.com/cheekybits/genny v1.0.0
> go: downloading github.com/marten-seemann/qtls-go1-15 v0.1.4
> go: downloading github.com/marten-seemann/qtls-go1-16 v0.1.3
> go: downloading github.com/marten-seemann/qtls-go1-17 v0.1.0-alpha.1
> go: downloading github.com/smallstep/nosql v0.3.6
> go: downloading howett.net/plist v0.0.0-20200419221736-3b63eb3a43b5
> go: downloading gopkg.in/square/go-jose.v2 v2.5.1
> go: downloading github.com/Masterminds/sprig/v3 v3.2.2
> go: downloading go.step.sm/cli-utils v0.2.0
> go: downloading github.com/urfave/cli v1.22.4
> go: downloading github.com/Masterminds/goutils v1.1.1
> go: downloading github.com/Masterminds/semver v1.4.2
> go: downloading github.com/huandu/xstrings v1.3.1
> go: downloading github.com/Masterminds/semver/v3 v3.1.1
> go: downloading github.com/imdario/mergo v0.3.11
> go: downloading github.com/mitchellh/copystructure v1.0.0
> go: downloading github.com/shopspring/decimal v1.2.0
> go: downloading github.com/spf13/cast v1.3.1
> go: downloading github.com/dgraph-io/badger v1.6.2
> go: downloading go.etcd.io/bbolt v1.3.5
> go: downloading github.com/dgraph-io/badger/v2 v2.0.1-rc1.0.20201003150343-5d1bab4fc658
> go: downloading github.com/go-sql-driver/mysql v1.5.0
> go: downloading github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e
> go: downloading github.com/manifoldco/promptui v0.8.0
> go: downloading github.com/cpuguy83/go-md2man v1.0.10
> go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.0
> go: downloading github.com/mitchellh/reflectwalk v1.0.0
> go: downloading github.com/dgraph-io/ristretto v0.0.4-0.20200906165740-41ebdbffecfd
> go: downloading github.com/dgryski/go-farm v0.0.0-20200201041132-a6ae2369ad13
> go: downloading github.com/samfoo/ansi v0.0.0-20160124022901-b6bd2ded7189
> go: downloading github.com/juju/ansiterm v0.0.0-20180109212912-720a0952cc2a
> go: downloading github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96
> go: downloading github.com/russross/blackfriday v1.5.2
> go: downloading github.com/golang/snappy v0.0.3
> go: downloading github.com/DataDog/zstd v1.4.5
> go: downloading github.com/russross/blackfriday/v2 v2.0.1
> go: downloading github.com/shurcooL/sanitized_anchor_name v1.0.0
> go: downloading github.com/mattn/go-colorable v0.1.6
> go: downloading github.com/mattn/go-isatty v0.0.12
> go: downloading github.com/lunixbochs/vtclean v1.0.0
> go get: added github.com/mholt/caddy-l4 v0.0.0-20210604203514-3cfcafe70883
> 2021/08/22 05:37:09 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/mholt/caddy-dynamicdns 
> go: downloading github.com/mholt/caddy-dynamicdns v0.0.0-20210719141002-2110627e7fea
> go: downloading github.com/NebulousLabs/go-upnp v0.0.0-20181203152547-b32978b8ccbf
> go: downloading gitlab.com/NebulousLabs/fastrand v0.0.0-20181126182046-603482d69e40
> go: downloading gitlab.com/NebulousLabs/go-upnp v0.0.0-20181011194642-3a71999ed0d3
> go get: added github.com/mholt/caddy-dynamicdns v0.0.0-20210719141002-2110627e7fea
> 2021/08/22 05:37:11 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/openstack-designate 
> go: downloading github.com/caddy-dns/openstack-designate v0.1.0
> go: downloading github.com/libdns/openstack-designate v0.1.0
> go: downloading github.com/gophercloud/gophercloud v0.15.0
> go get: added github.com/caddy-dns/openstack-designate v0.1.0
> 2021/08/22 05:37:12 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/azure 
> go: downloading github.com/caddy-dns/azure v0.2.0
> go: downloading github.com/libdns/azure v0.2.0
> go: downloading github.com/Azure/azure-sdk-for-go v52.4.0+incompatible
> go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
> go: downloading github.com/Azure/go-autorest/autorest v0.11.17
> go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.7
> go: downloading github.com/Azure/go-autorest/autorest/to v0.4.0
> go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.11
> go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.2
> go: downloading github.com/Azure/go-autorest/logger v0.2.0
> go: downloading github.com/dimchansky/utfbom v1.1.1
> go: downloading github.com/Azure/go-autorest/tracing v0.6.0
> go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
> go: downloading github.com/form3tech-oss/jwt-go v3.2.2+incompatible
> go: downloading github.com/mitchellh/go-homedir v1.1.0
> go: downloading github.com/Azure/azure-sdk-for-go v0.2.0-beta
> go: downloading github.com/form3tech-oss/jwt-go v1.0.2
> go get: added github.com/caddy-dns/azure v0.2.0
> 2021/08/22 05:37:19 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/vultr 
> go: downloading github.com/caddy-dns/vultr v0.0.0-20210105121117-3162aa6b9c27
> go: downloading github.com/libdns/vultr v0.0.0-20201128180404-1d5ee21ea62f
> go: downloading github.com/vultr/govultr/v2 v2.0.0
> go: downloading golang.org/x/oauth2 v0.0.0-20201109201403-9fd604954f58
> go: downloading github.com/google/go-querystring v1.0.0
> go: downloading github.com/hashicorp/go-retryablehttp v0.6.8
> go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
> go: downloading google.golang.org/appengine v1.6.7
> go get: added github.com/caddy-dns/vultr v0.0.0-20210105121117-3162aa6b9c27
> 2021/08/22 05:37:20 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/hetzner 
> go: downloading github.com/caddy-dns/hetzner v0.0.1
> go: downloading github.com/libdns/hetzner v0.0.1
> go get: added github.com/caddy-dns/hetzner v0.0.1
> 2021/08/22 05:37:20 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/digitalocean 
> go: downloading github.com/caddy-dns/digitalocean v0.0.0-20210809220558-ac6e4fd9e135
> go: downloading github.com/libdns/digitalocean v0.0.0-20210310230526-186c4ebd2215
> go: downloading github.com/digitalocean/godo v1.41.0
> go get: added github.com/caddy-dns/digitalocean v0.0.0-20210809220558-ac6e4fd9e135
> 2021/08/22 05:37:21 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/alidns 
> go: downloading github.com/caddy-dns/alidns v1.0.23
> go: downloading github.com/libdns/alidns v1.0.2-x2
> go get: added github.com/caddy-dns/alidns v1.0.23
> 2021/08/22 05:37:22 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/gandi 
> go: downloading github.com/caddy-dns/gandi v1.0.2
> go: downloading github.com/libdns/gandi v1.0.2
> go get: added github.com/caddy-dns/gandi v1.0.2
> 2021/08/22 05:37:22 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/duckdns 
> go: downloading github.com/caddy-dns/duckdns v0.3.1
> go: downloading github.com/libdns/duckdns v0.1.1
> go get: added github.com/caddy-dns/duckdns v0.3.1
> 2021/08/22 05:37:23 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/dnspod 
> go: downloading github.com/caddy-dns/dnspod v0.0.3
> go: downloading github.com/libdns/dnspod v0.0.3
> go: downloading github.com/nrdcg/dnspod-go v0.4.0
> go get: added github.com/caddy-dns/dnspod v0.0.3
> 2021/08/22 05:37:23 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/lego-deprecated 
> go: downloading github.com/caddy-dns/lego-deprecated v0.0.0-20210524165322-3ab114dafc6b
> go: downloading github.com/go-acme/lego/v4 v4.3.1
> go: downloading github.com/cpu/goacmedns v0.1.1
> go: downloading github.com/aliyun/alibaba-cloud-sdk-go v1.61.976
> go: downloading github.com/nrdcg/auroradns v1.0.1
> go: downloading github.com/labbsr0x/bindman-dns-webhook v1.0.2
> go: downloading github.com/cloudflare/cloudflare-go v0.14.0
> go: downloading github.com/nrdcg/desec v0.5.0
> go: downloading github.com/gophercloud/gophercloud v0.16.0
> go: downloading github.com/gophercloud/utils v0.0.0-20210216074907-f6de111f2eae
> go: downloading github.com/dnsimple/dnsimple-go v0.63.0
> go: downloading github.com/akamai/AkamaiOPEN-edgegrid-golang v1.1.0
> go: downloading github.com/exoscale/egoscale v0.46.0
> go: downloading cloud.google.com/go v0.70.0
> go: downloading google.golang.org/api v0.33.0
> go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
> go: downloading github.com/iij/doapi v0.0.0-20190504054126-0bbf12d6d7df
> go: downloading github.com/cenkalti/backoff/v4 v4.1.0
> go: downloading github.com/nrdcg/goinwx v0.8.1
> go: downloading github.com/pquerna/otp v1.3.0
> go: downloading github.com/aws/aws-sdk-go v1.37.27
> go: downloading github.com/linode/linodego v0.25.3
> go: downloading github.com/liquidweb/liquidweb-go v1.6.3
> go: downloading github.com/namedotcom/go v0.0.0-20180403034216-08470befbe04
> go: downloading github.com/nrdcg/namesilo v0.2.1
> go: downloading gopkg.in/ns1/ns1-go.v2 v2.4.4
> go: downloading github.com/oracle/oci-go-sdk v24.3.0+incompatible
> go: downloading github.com/ovh/go-ovh v1.1.0
> go: downloading github.com/sacloud/libsacloud v1.36.2
> go: downloading github.com/transip/gotransip/v6 v6.6.0
> go: downloading github.com/OpenDNS/vegadns2client v0.0.0-20180418235048-a3fa4a771d87
> go: downloading github.com/vultr/govultr/v2 v2.4.0
> go: downloading github.com/Azure/go-autorest/autorest/validation v0.1.0
> go: downloading github.com/labbsr0x/goh v1.0.1
> go: downloading gopkg.in/yaml.v2 v2.4.0
> go: downloading github.com/patrickmn/go-cache v2.1.0+incompatible
> go: downloading github.com/sirupsen/logrus v1.6.0
> go: downloading gopkg.in/ini.v1 v1.62.0
> go: downloading github.com/gofrs/uuid v3.2.0+incompatible
> go: downloading google.golang.org/grpc v1.38.0
> go: downloading github.com/fatih/structs v1.1.0
> go: downloading github.com/kolo/xmlrpc v0.0.0-20200310150728-e0350524596b
> go: downloading github.com/mitchellh/mapstructure v1.3.3
> go: downloading github.com/liquidweb/go-lwApi v0.0.5
> go: downloading github.com/go-resty/resty/v2 v2.1.1-0.20191201195748-d7b97669fe48
> go: downloading go.uber.org/ratelimit v0.0.0-20180316092928-c15da0234277
> go: downloading github.com/jmespath/go-jmespath v0.4.0
> go: downloading github.com/json-iterator/go v1.1.11
> go: downloading github.com/modern-go/reflect2 v1.0.1
> go: downloading github.com/go-errors/errors v1.0.1
> go: downloading github.com/konsorten/go-windows-terminal-sequences v1.0.3
> go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
> go: downloading go.opencensus.io v0.22.5
> go: downloading github.com/googleapis/gax-go/v2 v2.0.5
> go: downloading github.com/boombuler/barcode v1.0.1-0.20190219062509-6c824513bacc
> go: downloading github.com/liquidweb/liquidweb-cli v0.6.9
> go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
> go: downloading github.com/deepmap/oapi-codegen v1.3.11
> go: downloading github.com/jarcoal/httpmock v1.0.6
> go: downloading github.com/stretchr/testify v1.7.0
> go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
> go: downloading github.com/davecgh/go-spew v1.1.1
> go: downloading github.com/pmezard/go-difflib v1.0.0
> go: downloading github.com/stretchr/objx v0.3.0
> go: downloading github.com/k0kubun/go-ansi v0.0.0-20180517002512-3bf9e2903213
> go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
> go: downloading github.com/gofrs/uuid v1.2.0
> go: downloading github.com/oracle/oci-go-sdk v1.8.0
> go: downloading github.com/patrickmn/go-cache v1.0.0
> go get: added github.com/caddy-dns/lego-deprecated v0.0.0-20210524165322-3ab114dafc6b
> 2021/08/22 05:37:37 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/route53 
> go: downloading github.com/caddy-dns/route53 v1.1.2
> go: downloading github.com/libdns/route53 v1.1.1
> go get: added github.com/caddy-dns/route53 v1.1.2
> 2021/08/22 05:37:38 [INFO] exec (timeout=0s): /usr/local/go/bin/go get -d -v github.com/caddy-dns/cloudflare 
> go: downloading github.com/caddy-dns/cloudflare v0.0.0-20210607183747-91cf700356a1
> go: downloading github.com/libdns/cloudflare v0.1.0
> go get: added github.com/caddy-dns/cloudflare v0.0.0-20210607183747-91cf700356a1
> 2021/08/22 05:37:38 [INFO] Build environment ready
> 2021/08/22 05:37:38 [INFO] Building Caddy
> 2021/08/22 05:37:38 [INFO] exec (timeout=0s): /usr/local/go/bin/go mod tidy 
> go: downloading github.com/google/go-cmp v0.5.5
> go: downloading github.com/go-chi/chi v4.1.2+incompatible
> go: downloading gopkg.in/natefinch/lumberjack.v2 v2.0.0
> go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
> go: downloading github.com/klauspost/compress v1.13.0
> go: downloading github.com/smallstep/assert v0.0.0-20200723003110-82e2b9b3b262
> go: downloading github.com/alecthomas/chroma v0.9.1
> go: downloading github.com/naoina/toml v0.1.1
> go: downloading github.com/yuin/goldmark v1.3.7
> go: downloading github.com/yuin/goldmark-highlighting v0.0.0-20210516132338-9216f9c5aa01
> go: downloading github.com/BurntSushi/toml v0.3.1
> go: downloading gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15
> go: downloading github.com/Azure/go-autorest/autorest/mocks v0.4.1
> go: downloading gopkg.in/h2non/gock.v1 v1.0.15
> go: downloading github.com/hashicorp/go-hclog v0.9.2
> go: downloading github.com/golang/mock v1.5.0
> go: downloading github.com/onsi/ginkgo v1.14.0
> go: downloading github.com/onsi/gomega v1.10.1
> go: downloading github.com/urfave/cli v1.22.5
> go: downloading github.com/naoina/go-stringutil v0.1.0
> go: downloading github.com/kylelemons/godebug v1.1.0
> go: downloading github.com/rs/xid v1.2.1
> go: downloading github.com/alecthomas/assert v0.0.0-20170929043011-405dbfeb8e38
> go: downloading github.com/kr/pretty v0.2.1
> go: downloading github.com/smartystreets/goconvey v1.6.4
> go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
> go: downloading github.com/uber-go/atomic v1.3.2
> go: downloading github.com/jmespath/go-jmespath/internal/testify v1.5.1
> go: downloading github.com/h2non/parth v0.0.0-20190131123155-b4df798d6542
> go: downloading github.com/dlclark/regexp2 v1.4.0
> go: downloading github.com/alecthomas/colour v0.0.0-20160524082231-60882d9e2721
> go: downloading github.com/alecthomas/repr v0.0.0-20180818092828-117648cd9897
> go: downloading github.com/sergi/go-diff v1.0.0
> go: downloading github.com/kr/text v0.1.0
> go: downloading github.com/jtolds/gls v4.20.0+incompatible
> go: downloading github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d
> go: downloading github.com/nxadm/tail v1.4.4
> go: downloading github.com/chzyer/test v0.0.0-20180213035817-a1ea475d72b1
> go: downloading github.com/danwakefield/fnmatch v0.0.0-20160403171240-cbb64ac3d964
> go: downloading github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
> go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
> go: downloading github.com/mattn/go-colorable v0.1.8
> go: downloading github.com/chzyer/logex v1.1.10
> go: downloading github.com/OneOfOne/xxhash v1.2.2
> go: downloading github.com/spaolacci/murmur3 v1.1.0
> go: downloading github.com/fsnotify/fsnotify v1.4.9
> 2021/08/22 05:37:41 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /usr/bin/caddy -ldflags -w -s -trimpath 
> 2021/08/22 05:41:04 [INFO] Build complete: ./caddy
> 2021/08/22 05:41:04 [INFO] Skipping cleanup as requested; leaving folder intact: /tmp/buildenv_2021-08-22-0536.187304441
> Removing intermediate container 665f943cd013
>  ---> 34fe088c60e3
> Step 3/10 : FROM caddy:builder-alpine
>  ---> f64009d1a35a
> Step 4/10 : COPY --from=builder /usr/bin/caddy /usr/bin/caddy
>  ---> 488741f3929c
> Step 5/10 : RUN apk update &&     apk add --no-cache --virtual ca-certificates caddy tor wget &&     mkdir /v2ray &&     wget -qO- https://github.com/v2fly/v2ray-core/releases/latest/download/v2ray-linux-64.zip | busybox unzip - &&     chmod +x /v2ray &&     rm -rf /var/cache/apk/* &&     apk del .build-deps
>  ---> Running in d76662be9822
> fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
> fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
> v3.14.1-64-g22f341e695 [https://dl-cdn.alpinelinux.org/alpine/v3.14/main]
> v3.14.1-63-gb2aec05acf [https://dl-cdn.alpinelinux.org/alpine/v3.14/community]
> OK: 14934 distinct packages available
> fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
> fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
> (1/11) Installing caddy (2.4.3-r0)
> Executing caddy-2.4.3-r0.pre-install
> (2/11) Installing libcap (2.50-r0)
> (3/11) Installing libevent (2.1.12-r2)
> (4/11) Installing xz-libs (5.2.5-r0)
> (5/11) Installing libseccomp (2.5.1-r2)
> (6/11) Installing zstd-libs (1.4.9-r1)
> (7/11) Installing tor (0.4.5.10-r0)
> Executing tor-0.4.5.10-r0.pre-install
> (8/11) Installing libunistring (0.9.10-r1)
> (9/11) Installing libidn2 (2.3.1-r0)
> (10/11) Installing wget (1.21.1-r1)
> (11/11) Upgrading ca-certificates (20191127-r5 -> 20210822.054221)
> Executing busybox-1.33.1-r3.trigger
> OK: 75 MiB in 31 packages
> Archive:  -
>    creating: systemd/
>    creating: systemd/system/
>   inflating: systemd/system/v2ray@.service
>   inflating: systemd/system/v2ray.service
>   inflating: config.json
>   inflating: v2ctl
>   inflating: geosite.dat
>   inflating: vpoint_socks_vmess.json
>   inflating: vpoint_vmess_freedom.json
>   inflating: geoip.dat
>   inflating: v2ray
> WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/main: No such file or directory
> WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/community: No such file or directory
> ERROR: No such package: .build-deps
> The command '/bin/sh -c apk update &&     apk add --no-cache --virtual ca-certificates caddy tor wget &&     mkdir /v2ray &&     wget -qO- https://github.com/v2fly/v2ray-core/releases/latest/download/v2ray-linux-64.zip | busybox unzip - &&     chmod +x /v2ray &&     rm -rf /var/cache/apk/* &&     apk del .build-deps' returned a non-zero code: 1
> ```

删.build-deps是为了减少占用空间的

DaoChen6 commented:
```/bin/sh: ENV: not found
The command '/bin/sh -c apk update &&     apk add --no-cache --virtual ca-certificates caddy tor wget curl &&     mkdir /v2ray &&     wget -qO- https://github.com/v2fly/v2ray-core/releases/latest/download/v2ray-linux-64.zip | busybox unzip - &&     chmod +x /v2ray &&     rm -rf /var/cache/apk/* && ENV XDG_CONFIG_HOME /etc/caddy' returned a non-zero code: 127```

DaoChen6 commented:
``/bin/sh: ENV: not found
The command '/bin/sh -c apk update &&     apk add --no-cache --virtual ca-certificates caddy tor wget curl &&     mkdir /v2ray &&     wget -qO- https://github.com/v2fly/v2ray-core/releases/latest/download/v2ray-linux-64.zip | busybox unzip - &&     chmod +x /v2ray &&     rm -rf /var/cache/apk/* && ENV XDG_CONFIG_HOME /etc/caddy' returned a non-zero code: 127``
